src/otx/core/model/module/visual_prompting.py

```diff
@@ -211,3 +237,108 @@ def _inference_step(self, metric: MetricCollection, inputs: VisualPromptingBatch
+        batch_idx: int,
+    ) -> Tensor:
+        """Skip training_step unused in zero-shot visual prompting."""
+        self.model(inputs)
+
+    def validation_step(
+        self,
+        inputs: VisualPromptingBatchDataEntity | ZeroShotVisualPromptingBatchDataEntity,
+        batch_idx: int,
+    ) -> None:
+        """Skip validation_step unused in zero-shot visual prompting."""
+
+    def _inference_step(
+        self,
+        metric: MetricCollection,
+        inputs: VisualPromptingBatchDataEntity | ZeroShotVisualPromptingBatchDataEntity,
+        batch_idx: int,
+    ) -> None:
+        """Perform a single inference step on a batch of data from the inference set."""
+        preds = self.model(inputs)
+
+        if not isinstance(preds, ZeroShotVisualPromptingBatchPredEntity):
+            raise TypeError(preds)
+
+        converted_entities = self._convert_pred_entity_to_compute_metric(preds, inputs)
+        for _name, _metric in metric.items():
+            if _name == "mAP":
+                # MeanAveragePrecision
+                _preds = [
+                    {
+                        k: v > 0.5 if k == "masks" else v.squeeze(1).to(self.device) if k == "labels" else v
+                        for k, v in ett.items()
+                    }
+                    for ett in converted_entities["preds"]
+                ]
+                _target = converted_entities["target"]
+
+                # match #_preds and #_target
+                if len(_preds) > len(_target):
+                    # interpolate _target
+                    num_diff = len(_preds) - len(_target)
+                    for i in range(num_diff):
+                        _target.append(_target[i])
+                elif len(_preds) < len(_target):
+                    num_diff = len(_target) - len(_preds)
+                    for i in range(num_diff):
+                        _preds.append(_preds[i])
+
+                _metric.update(preds=_preds, target=_target)
+            elif _name in ["IoU", "F1", "Dice"]:
+                # BinaryJaccardIndex, BinaryF1Score, Dice
+                for cvt_preds, cvt_target in zip(converted_entities["preds"], converted_entities["target"]):
+                    _metric.update(
+                        cvt_preds["masks"].sum(dim=0).clamp(0, 1),
+                        cvt_target["masks"].sum(dim=0).clamp(0, 1),
+                    )
+
+    def state_dict(self) -> dict[str, Any]:
+        """Return state dictionary of model entity with reference features, masks, and used indices."""
+        state_dict = super().state_dict()
+        state_dict.update(
+            {
+                "model.model.reference_feats": self.model.model.reference_feats,
+                "model.model.reference_masks": self.model.model.reference_masks,
+                "model.model.used_indices": self.model.model.used_indices,
+            },
+        )
```

Comment:
Can we move this to `OTXZeroShotVisualPromptingModel` and `load_state_dict()` too? `OTXModel` should be able to save and load its state dict as standalone.

Reply:
Done at https://github.com/openvinotoolkit/training_extensions/pull/2834/commits/9ba3007e76ade8212ee9f62b04ddceb71a846bc0.